frontend/src/textToSpeech/textToSpeech.js

```diff
@@ -0,0 +1,380 @@
+import React, { Component } from 'react';
+import * as PropTypes from 'prop-types';
+
+import { Footer, Navbar, LoadingPage } from '../UILibrary/components';
+
+const playButton = (color) => {
+    return (
+        <svg
+            xmlns = "http://www.w3.org/2000/svg"
+            viewBox = "0 0 26 26"
+            fill = {color}
+            className = "bi bi-play-button"
+        >
+            <polygon
+                className="play-btn__svg"
+                points="9.33 6.69 9.33 19.39 19.3 13.04 9.33 6.69"
+            />
+            <path
+                d={'M26,13A13,13,0,1,1,13,0,13,13,0,0,1,26,13ZM13,2.18A10.89,10.89'
+                + ',0,1,0,23.84,13.06,10.89,10.89,0,0,0,13,2.18Z'}
+            />
+        </svg>
+    );
+};
+
+export class TextToSpeech extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            textData: null,
+            sentenceIndex: 0,
+            showModal: false,
+            userText: '',
+            graded: false,
+            grade: null,
+            continue: false,
+            showAnswer: false,
+        };
+        this.modalHandler = this.modalHandler.bind(this);
+    }
+
+    componentDidMount = async () => {
+        const apiURL = `/api/get_indiv_sentences/${this.props.textID}`;
+        const response = await fetch(apiURL);
+        const data = await response.json();
+        this.setState({
+            textData: data,
+        });
+        console.log(this.state.textData);
+        document.addEventListener('keydown', this.handleKeyDown, true);
+    }
+
+    componentWillUnmount() {
+        document.removeEventListener('keydown', this.handleKeyDown, false);
+    }
+
+    getCurrentSentence = () => {
+        return this.state.textData[this.state.sentenceIndex]['sentence'];
+    }
+
+    changeSentence = (delta) => {
+        const { textData, sentenceIndex } = this.state;
+        const listLength = textData.length;
+        const newsentenceIndex = (sentenceIndex + delta + listLength) % listLength;
+        this.setState({
+            sentenceIndex: newsentenceIndex,
+            showNext: true,
+        });
+        this.reset();
+    }
+
+    reset = async () => {
+        this.setState({
+            showModal: false,
+            userText: '',
+            graded: false,
+            grade: null,
+            continue: false,
+            showAnswer: false,
+        });
+        document.getElementById('content').value = '';
```

Comment:
This and the other places where we manually grab the value of the `textarea` with `id="content"` stand out to me. At least as far as I can tell so far, a more conventional approach would be to have an `onChange={}` listener on the `<textarea>` that sets the `event.target.value` to `state.userText`. That way, you always have an up-to-date match of what the user is typing rather than having to use vanilla JavaScript `document.getElementById` to reach into the DOM and manually grab the value.

Did you consider that more conventional approach? Did you encounter any problems with it?

In general, the approach you've used here mixes React's [declarative style with a more imperative style](https://medium.com/@myung.kim287/declarative-vs-imperative-251ce99c6c44), and in general it benefits both development and the user experience to stick to the former.